### PR TITLE
Updates TimeSeriesMeta documentation.

### LINKED ISSUE #3651: Bugs in the TimeSeriesMeta creation.
### Description

When passing parameters to ``TimeSeriesMetaData``, if only ``meta`` and ``timerange`` are provided, the returned object is empty, and doesn't even store the ``meta`` or ``timerange`` passed.
The docs stating that these parameters are optional are wrong and need to be changed.
All parameters need to be passed for object creation.
### Expected behavior

When all parameters are passed, the object is created and returned. Otherwise, raise an error.
### Actual behavior

While a TimeSeriesMeta is returned when just ``meta`` or ``timerange`` is passed, it is empty, and doesn't even store the meta passed.

### Steps to Reproduce

Create the ``TimeSeriesMetaData`` object by calling  ``sunpy.timeseries.TimeSeriesMetaData()``
just passing ``meta`` and ``timerange``
### System Details

  - SunPy Version: sunpy 1.0.6
  - Astropy Version: astropy 4.0 
  - Python Version: python 3.7 
  - OS information: ububtu 18.04

As per the discussion in #3639

### PR DESCRIPTION
**Pull request by Raahul-Singh**
### Description

Fixes #3651

### CODE REVIEW

File: changelog/3680.doc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Updates TimeSeriesMeta docs to make passing all parameters mandatory.
```
Comment (nabobalis) [inline changelog/3680.doc.rst]:
```suggestion
Corrected misleading `~sunpy.timeseries.metadata.TimeSeriesMetaData` documentation about optional parameters.
```

**Comment by Raahul-Singh** [comment]
@hayesla This fixes the docs. I tried making the parameters compulsory in the code as well but that resulted in a lot of tests failing as they were missing many parameters while making the `TimeSeriesMeta` object.
May I make these changes and change all the tests where they omit parameters?

**Comment by nabobalis** [comment]
According to the ``__init__``, it does  seem all you need is a time range for the function to return something (even if its empty). I am unsure if this doc change is technically correct. Hopefully someone with more experience with time series can chime in.

**Comment by Raahul-Singh** [comment]
> According to the `__init__`, it does seem all you need is a time range for the function to return something (even if its empty). I am unsure if this doc change is technically correct. Hopefully someone with more experience with time series can chime in.

This is as per this comment. 
https://github.com/sunpy/sunpy/pull/3639#issuecomment-571322842

**Comment by Raahul-Singh** [comment]
> I've listed the possible signatures in #3709 - I think these parameters are optional, and to fix #3639 some checks need to be added to the code in `__init__`. I think the intention is that if `meta` is a tuple it should be a tuple of `(timerange, colnames, meta)`, and if it's a list it should be a list of these tuples. @Raahul-Singh do you fancy trying to add these extra checks to the code?

Sure, I'll work on it!

**Comment by nabobalis** [comment]
No I think they can be added without waiting on that PR.

**Comment by Raahul-Singh** [comment]
Is anything more required here?

**Comment by nabobalis** [comment]
Just needs reviews.

**Comment by Raahul-Singh** [comment]
Thank You @nabobalis, @dstansby. This was the first issue I opened and successfully closed on SunPy! :smile:

**Comment by dstansby** [comment]
Congratulations!

**Pull request merged**